src/Raven.Server/Documents/Queries/StreamJsonlDocumentQueryResultWriter.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Raven.Server.Json;
+using Sparrow.Json;
+
+namespace Raven.Server.Documents.Queries
+{
+    public class StreamJsonlDocumentQueryResultWriter : IStreamQueryResultWriter<Document>
+    {
+        private readonly AsyncBlittableJsonTextWriter _writer;
+        private readonly JsonOperationContext _context;
+
+        public StreamJsonlDocumentQueryResultWriter(Stream stream, JsonOperationContext context)
+        {
+            _context = context;
+            _writer = new AsyncBlittableJsonTextWriter(context, stream);
+        }
+
+        public ValueTask DisposeAsync()
+        {
+            return _writer.DisposeAsync();
+        }
+
+        public void StartResponse()
+        {
+        }
+
+        public void StartResults()
+        {
+        }
+
+        public void EndResults()
+        {
+        }
+
+        public async ValueTask AddResultAsync(Document res, CancellationToken token)
+        {
+            _writer.WriteDocument(_context, res, metadataOnly: false);
+            _writer.WriteNewLine();
+            await _writer.MaybeFlushAsync(token);
+        }
+
+        public void EndResponse()
+        {
+        }
+
+        public ValueTask WriteErrorAsync(Exception e)
+        {
+            throw new NotSupportedException();
+        }
+
+        public ValueTask WriteErrorAsync(string error)
+        {
+            throw new NotSupportedException();
```

Comment:
Note that errors should be in done in a way that would be _breaking_ for the client.
What we want to ensure is that there is no way that they will accept that as a valid value.

Reply:
Can there be a mix of errors and results, or if there is an error then the only thing in the response is the error?

Reply:
If there is an error, then we report that and abort the rest of the operation